### PR TITLE
Refactor company create action to move fatty code to service

### PR DESCRIPTION
**Pull request by shalapatil**
## Notion card
https://www.notion.so/Refactor-controllers-51ad6bf864c448308db7a9c6f65241ad

## Summary
Moved fatty controller code to a service.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

File: app/services/create_company_service.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  attr_accessor :company
+
+  def initialize(params, current_user)
+    @params = params
+    @current_user = current_user
+    @company = nil
+  end
+
+  def process
+    @company = Company.new(params)
```
[the remark this answers is not in the document]
Comment (shalapatil) [reply app/services/create_company_service.rb]: done

File: app/services/create_company_service.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    company.save!
+    update_current_user
```
[the remark this answers is not in the document]
Comment (shalapatil) [reply app/services/create_company_service.rb]: done

**Comment by shalapatil** [comment]
@apoorv-mishra @supriya3105 Please test the following workflow
Create new company, see if user roles is correct.

**Comment by apoorv1316** [comment]
@shalapatil After sign up app breaking on staging. cc @supriya3105 
Screenshot 2023-01-02 at 8 45 47 PM

**Pull request merged**